src/Integration/Binding/IRuleSetReferenceChecker.cs

```diff
@@ -25,10 +25,15 @@ namespace SonarLint.VisualStudio.Integration.Binding
     internal interface IRuleSetReferenceChecker
     {
         /// <summary>
-        /// Return true if the project's ruleSets directly include the target ruleSet, otherwise false
+        /// Return true if the RuleSet declaration directly includes the target ruleSet, otherwise false
         /// </summary>
         /// <remarks>We don't currently check nested includes in RuleSet
         /// i.e. if Project has RuleSet A which includes RuleSet B that includes RuleSet C, IsReferenced(Project, C) returns false.</remarks>
-        bool IsReferenced(Project project, string targetRuleSetFilePath);
+        bool IsReferenced(Project project, RuleSetDeclaration declaration, string targetRuleSetFilePath);
```

Comment:
I realise you've just made a previously private method public and added it to the interface, but the method signature doesn't look right - it shouldn't be necessary to provide both the `project` and a `ruleSetDeclaration` since they are related objects.

However, I think this is just exposing the underlying problem
`ISolutionRuleSetsInformationProvider.TryGetProjectRuleSetFilePath`, which has a very similar signature.

The `ruleSetDeclaration` actually contains enough information to carry out the check without the supplying the `project` too.

I think we should fix that problem separately.

Reply:
Yep, I see now that ruleSetDeclaration has RuleSetProjectFullName.
I will do it in a separate PR.

Reply:
There are a number of possible fixes:
1) remove the `project` parameter, and just pass in the `rulesetDeclaration` and target ruleset path.
2) change the `rulesetDeclaration` parameter to `sourceRuleSetFilePath`,
3) change the signature completely to be more general
e.g. `bool TryResolveRuleSetFilePath(string sourceRuleSetPath, out string fullFilePath, params string[] searchPaths)`
I'm not sure which option is best - it depends what makes sense on the interface i.e. logically what operation do we want to expose? (and it also depends on the amount of work involved).

Reply:
I think removing the project parameter is simplest:
https://github.com/SonarSource/sonarlint-visualstudio/pull/1438